addons/mail/static/src/models/channel_member_list_view.js

```diff
@@ -1,32 +1,63 @@
 /** @odoo-module **/
 
 import { registerModel } from '@mail/model/model_core';
-import { one } from '@mail/model/model_field';
-import { clear, replace } from '@mail/model/model_field_command';
+import { attr, one } from '@mail/model/model_field';
+import { clear, replace, insert} from '@mail/model/model_field_command';
 
 registerModel({
     name: 'ChannelMemberListView',
     identifyingFields: [['chatWindowOwner', 'threadViewOwner']],
+    lifecycleHooks: {
+        _created() {
+            this.fetchChannelMembers();
+        }
+    },
     recordMethods: {
+        /**
+         * Handles click on the "load more members" button.
+         */
+        async onClickLoadMoreMembers() {
+            this.fetchChannelMembers();
+        },
+        async fetchChannelMembers() {
+            const channelMembersList = await this.messaging.rpc({
+                model: 'mail.channel',
+                method: 'load_more_members',
+                args: [[this.channel.id]],
+                kwargs: {
+                    'known_member_ids': this.channel.channelMembers.map(channelMember => channelMember.partner.id),
+                },
+            });
+            if (!this.channel) {
+                return;
+            }
+            this.channel.update({ memberCount: channelMembersList.count });
+            for (const partner of channelMembersList.partners) {
+                const relation = { channel: replace(this.channel) };
+                relation.partner = insert(partner);
+                this.channel.update({ channelMembers: insert(relation) });
+            }
+        },
```

Comment:
And just to make changes complete, instead of `insert(AndLink)`, use `insertAndReplace`:
```js
this.channel.update({
    channelMembers: insertAndReplace(
        channelMembersList.partners.map(
            partner => ({
                channel: replace(this.channel),
                partner: replace(partner),
            })
        )
    )
});
```

Instead of the iterative `for-of` loop.

Reply:
This will likely be completely refactored when I will add guest in an other PR.
It also doesn't work. For some reason, it throw an `record.exists is not a function`.